### PR TITLE
[#795] Export Metadata Subfield Subjects

### LINKED ISSUE #795: Export Metadata Subfield Subjects
Export the Metadata subfield [Subjects](https://inveniordm.docs.cern.ch/reference/metadata/#subjects-0-n)

### PR DESCRIPTION
**Pull request by Meowcenary**
closes #795

### CODE REVIEW

File: app/controllers/custom_authorities_controller.rb [hunk 1]
```diff
@@ -15,15 +15,15 @@ def query_mesh
     render json: hits
   end
 
-  def lcsh_names
+  def lcnaf_names
```
Comment (Meowcenary) [inline app/controllers/custom_authorities_controller.rb:18]: Changing this to reflect end point and what is being retrievd.

File: app/models/concerns/galtersufia/generic_file/header_lookup.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        subject_id_uris[subject_match_index].split('/').last
+      end
+
+      def hits_from_mesh_sparql(json_parsed_result)
+        json_parsed_result.try(:[], "results").try(:[], "bindings")
+      end
+
+      def pid_from_mesh_hits(id_uris)
+        # take the first match found, split it's URI to get the PID at the end
+        id_uris.shift["d"]["value"].split('/').last
```
Comment (Meowcenary) [inline app/models/concerns/galtersufia/generic_file/header_lookup.rb:78]: I tested this a bit and the mesh hits matched better than the lcsh hits for whatever reason. It seems kind of pointless to sift through a list if we dont have to, but I guess we'll find out in testing if my understanding of this is incorrect.

File: app/models/invenio_rdm_record_converter.rb [hunk 3]
```diff
@@ -46,6 +49,11 @@ def initialize(generic_file=nil)
     # @permissions = permissions(generic_file)
   end
 
+  def to_json(options={})
```
Comment (Meowcenary) [inline app/models/invenio_rdm_record_converter.rb:52]: As I mentioned before, this is necessary to pass the options hash which in turn makes it so we can exclude certain variables from the object when we convert it to JSON format.

File: config/routes.rb [hunk 4]
```diff
@@ -16,7 +16,7 @@
   get '/authorities/generic_files/creator' => 'custom_authorities#query_users'
   get '/authorities/generic_files/contributor' => 'custom_authorities#query_users'
   get '/authorities/generic_files/verify_user' => 'custom_authorities#verify_user'
-  get '/authorities/generic_files/subject_name' => 'custom_authorities#lcsh_names'
+  get '/authorities/generic_files/subject_name' => 'custom_authorities#lcnaf_names'
```
Comment (Meowcenary) [inline config/routes.rb:19]: Update to match the renamed controller method.

File: spec/controllers/custom_authorities_controller_spec.rb [hunk 5]
```diff
@@ -195,21 +195,21 @@
     end
   end # query_users
 
-  describe '#lcsh_names', :vcr do
+  describe '#lcnaf_names', :vcr do
```
Comment (Meowcenary) [inline spec/controllers/custom_authorities_controller_spec.rb:198]: Updating this spec for renamed method and route.

**Comment by Meowcenary** [comment]
This is now updated and good to go. Hopefully this PR is much easier to read.

File: app/models/concerns/galtersufia/generic_file/header_lookup.rb (hunk 2 above)
Comment (carsonicator) [reply app/models/concerns/galtersufia/generic_file/header_lookup.rb:78]: Just curious...so the ORM is translating requests into SPARQL queries?

File: spec/controllers/custom_authorities_controller_spec.rb (hunk 5 above)
Comment (sharpattack) [reply spec/controllers/custom_authorities_controller_spec.rb:198]: Great! Thanks for doing this.

**Review by sharpattack — APPROVED** [review]
Thanks for updating. This PR does not look much different than my previous review. If the full spec was green, this is good to me.

**Review by carsonicator — COMMENTED** [review]
Thanks, @Meowcenary

**Pull request merged**